tests/jballerina-unit-test/src/test/resources/test-src/record/open_record.bal

```diff
@@ -473,3 +473,29 @@ function addStringToMapValue(string s) returns string {
     mapValue = mapValue + s;
     return mapValue;
 }
+
+public type CustomDetail record {
+    string message;
+    error cause?;
+};
+
+public const FOO_REASON = "FooError";
+public type FooError error<FOO_REASON, CustomDetail>;
+
+public const BAR_REASON = "BarError";
+public type BarError error<BAR_REASON, CustomDetail>;
+
+public type Error FooError|BarError;
+
+public type MyRecord record {
+    typedesc<Error>[] myErrorTypes = [FooError];
+};
+
+function testCustomErrorTypeDescFieldOnRecord() {
+    MyRecord m = {};
+    typedesc<Error> e = m.myErrorTypes[0];
+    if(!(e is typedesc<FooError>)) {
+        error panicError = error("expected typedesc<FooError> but found: " + e.toString());
+        panic panicError;
+    }
```

Comment:
Shall we use a simple string as the reason and move the current reason you've used as the message field of the detail?

```suggestion
    if !(e is typedesc<FooError>) {
        panic error("AssertionError", message = "expected typedesc<FooError> but found: " + e.toString());
    }
```

Reply:
+1, fixed.